protobuf/lib/src/protobuf/field_set.dart

```diff
@@ -390,6 +392,46 @@ class _FieldSet {
     return _getDefaultMap<K, V>(_nonExtensionInfoByIndex(index));
   }
 
+  /// The implementation of a generated getter for `bool` fields.
+  bool _$getB(int index, bool defaultValue) {
+    var value = _values[index];
+    if (value == null) {
+      if (defaultValue != null) return defaultValue;
+      value = _getDefault(_nonExtensionInfoByIndex(index));
+    }
+    bool result = value;
+    return result;
+  }
+
+  /// The implementation of a generated getter for `bool` fields that default to
+  /// `false`.
+  bool _$getBF(int index) {
```

Comment:
Could we reduce duplication with something like:
```dart
  T _getWithDefault<T>(int index, T default) {
    var value = _values[index];
    T result = value ?? default;
    return result;
  }

  _getIntegerDefaultZero(int index) => _getWithDefault(index, 0);
  _getStringDefaultEmpty(int index) => _getWithDefault(index, '');
  ...
```

Reply:
@sigurdm  I would not do that - casts / checks of generic type are more expensive that casts to a fixed type because casts to a fixed type can be specialized. This is especially true of primitive types like bool, num, String where the specialized JavaScript predicates JIT to a few instructions.

Knowing the default also allows dart2js to potentially generate better code for ??.

Reply:
What a shame. But good to know!